internal/blobtesting/map.go

```diff
@@ -17,11 +17,15 @@ import (
 // DataMap is a map of blob ID to their contents.
 type DataMap map[blob.ID][]byte
 
+// versionedDataMap is a map of blob ID to a slice with versions of contents.
+type versionedDataMap map[blob.ID][][]byte
```

Comment:
Should there be a separate `versionedMapStorage` implementation that is used only for versioned store tests? or use a single implementation (as proposed here) in all cases?

Either way, the changes related to `blobtesting/map` become a good candidate for a separate, self-contained PR. We can make those changes independently, make sure it does not break existing testing, merge upstream and iterate.

Reply:
agreed, I tried to put something together quickly, but this really belongs is a dedicated PR